test/test_corpus.cc

```diff
@@ -495,6 +495,129 @@ TEST_P(ExpectationTest, PerPhaseTest) { // NOLINT
     // Allow later phases to have errors that we didn't test for
     errorQueue->drainAllErrors();
 
+    // now we test the incremental resolver
+
+    auto disableStressIncremental =
+        BooleanPropertyAssertion::getValue("disable-stress-incremental", assertions).value_or(false);
+    auto isAutogenTest = test.expectations.find("autogen") != test.expectations.end();
+    if (disableStressIncremental || isAutogenTest) {
+        TEST_COUT << "errors OK" << '\n';
+        return;
+    }
+
+    got.clear();
+    auto symbolsBefore = gs->symbolsUsed();
+
+    vector<ast::ParsedFile> newTrees;
+    for (auto &f : trees) {
+        const int prohibitedLines = f.file.data(*gs).source().size();
+        auto newSource = fmt::format("{}\n{}", string(prohibitedLines, '\n'), f.file.data(*gs).source());
+        auto newFile =
+            make_shared<core::File>((string)f.file.data(*gs).path(), move(newSource), f.file.data(*gs).sourceType);
+        gs = core::GlobalState::replaceFile(move(gs), f.file, move(newFile));
+
+        unique_ptr<KeyValueStore> kvstore;
+        // this replicates the logic of pipeline::indexOne
+
+        auto nodes = parser::Parser::run(*gs, f.file);
+        auto expectation = test.expectations.find("parse-tree");
+        if (expectation != test.expectations.end()) {
```

Comment:
btw, all of these could be `test.expectations.contains("parse-tree")` if you want to make this more concise...